src/diffusers/models/unet_2d_condition.py

```diff
@@ -123,20 +124,45 @@ def __init__(
         num_class_embeds: Optional[int] = None,
         upcast_attention: bool = False,
         resnet_time_scale_shift: str = "default",
+        time_embed_dim: int = None,
+        time_embedding_type: str = "positional", # fourier, positional
+        timestep_act_2: bool = False, 
+        class_labels_dim: Optional[int] = None,
+        conv_in_kernel: int = 3,
+        conv_out_kernel: int = 3,
+        norm_out: bool = True,
+        attn1_types: Tuple[bool, None]=(None,'self','self','self'),
+        attn2_types: Tuple[bool, None] = (None, 'cross','cross','cross'),
+        downsample: Optional[Tuple[bool]] = None,
+        upsample: Optional[Tuple[bool]] = None,
+        skip_freq: str = "layer" # layer, block
     ):
         super().__init__()
 
         self.sample_size = sample_size
-        time_embed_dim = block_out_channels[0] * 4
+        self.skip_freq = skip_freq
+        if time_embed_dim is None:
+            time_embed_dim = block_out_channels[0] * 4
 
         # input
-        self.conv_in = nn.Conv2d(in_channels, block_out_channels[0], kernel_size=3, padding=(1, 1))
+        conv_in_padding = (conv_in_kernel-1)//2
+        self.conv_in = nn.Conv2d(in_channels, block_out_channels[0], kernel_size=conv_in_kernel, padding=conv_in_padding)
 
         # time
-        self.time_proj = Timesteps(block_out_channels[0], flip_sin_to_cos, freq_shift)
-        timestep_input_dim = block_out_channels[0]
-
-        self.time_embedding = TimestepEmbedding(timestep_input_dim, time_embed_dim)
+        if time_embedding_type == "fourier":
```

Comment:
actually can we derive the `time_embed_dim` automatically to begin with if we have a `GaussianFourierProjection` ? It's probably dependent on something else no? I'd even suggest to hardcode it for now if we have to